core/consensus/validation/babe_block_validator.cpp

```diff
@@ -93,92 +72,66 @@ namespace kagome::consensus {
+      const primitives::BlockHeader &header,
+      const primitives::AuthorityId &authority_id,
+      const Threshold &threshold,
+      const Randomness &randomness) const {
+    log_->debug("Validates block signed by authority: {}",
+                authority_id.id.toHex());
 
-  outcome::result<std::pair<Seal, BabeBlockHeader>>
-  BabeBlockValidator::getBabeDigests(const primitives::Block &block) const {
-    // valid BABE block has at least two digests: BabeHeader and a seal
-    if (block.header.digest.size() < 2) {
-      log_->info(
-          "valid BABE block must have at least 2 digests, this one have {}",
-          block.header.digest.size());
-      return ValidationError::INVALID_DIGESTS;
-    }
-    const auto &digests = block.header.digest;
+    // get BABE-specific digests, which must be inside of this block
+    OUTCOME_TRY(babe_digests, getBabeDigests(header));
+    auto [seal, babe_header] = babe_digests;
 
-    // last digest of the block must be a seal - signature
-    auto seal_res = getFromVariant<primitives::Seal>(digests.back());
-    if (!seal_res) {
-      log_->info("last digest of the block is not a Seal");
-      return ValidationError::INVALID_DIGESTS;
+    // signature in seal of the header must be valid
+    if (!verifySignature(header, babe_header, seal, authority_id.id)) {
+      return ValidationError::INVALID_SIGNATURE;
     }
 
-    OUTCOME_TRY(babe_seal_res, scale::decode<Seal>(seal_res.value().data));
-
-    for (const auto &digest :
-         gsl::make_span(digests).subspan(0, digests.size() - 1)) {
-      if (auto consensus_dig = getFromVariant<primitives::PreRuntime>(digest);
-          consensus_dig) {
-        if (auto header = scale::decode<BabeBlockHeader>(consensus_dig->data);
-            header) {
-          // found the BabeBlockHeader digest; return
-          return {babe_seal_res, header.value()};
-        }
-      }
+    // VRF must prove that the peer is the leader of the slot
+    if (!verifyVRF(babe_header, authority_id.id, threshold, randomness)) {
+      return ValidationError::INVALID_VRF;
     }
 
-    log_->warn("there is no BabeBlockHeader digest in the block");
-    return ValidationError::INVALID_DIGESTS;
+    // peer must not send two blocks in one slot
+    //    if (!verifyProducer(babe_header)) {
+    //      return ValidationError::TWO_BLOCKS_IN_SLOT;
+    //    }
+    return outcome::success();
   }
 
   bool BabeBlockValidator::verifySignature(
-      const primitives::Block &block,
+      const primitives::BlockHeader &header,
       const BabeBlockHeader &babe_header,
       const Seal &seal,
-      gsl::span<const primitives::Authority> authorities) const {
+      const primitives::SessionKey &public_key) const {
     // firstly, take hash of the block's header without Seal, which is the last
     // digest
-    auto block_copy = block;
-    block_copy.header.digest.pop_back();
+    auto header_copy = header;
```

Comment:
```suggestion
    auto unsealed_header = header;
```
Just so it's immediately clear why make a copy